operators/pkg/controller/common/user/utils.go

```diff
@@ -0,0 +1,37 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package user
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GetSecret gets the first secret in a list that matches the namespace and the name.
+func GetSecret(list corev1.SecretList, namespace, name string) *corev1.Secret {
+	for _, secret := range list.Items {
+		if secret.Namespace == namespace && secret.Name == name {
+			return &secret
+		}
+	}
+	return nil
+}
+
+// ChecksUser checks that a secret contains the required fields expected by the user reconciler.
+func ChecksUser(t *testing.T, secret *corev1.Secret, expectedUsername string, expectedRoles []string) {
```

Comment:
Does this introduce a dependency to  `testing` in production code?

Reply:
Good question !
Functions in `*_test.go` files are not available for import by other tests. Looks like there is no easy way to share some code between tests.

I did some tests and it looks like there is not dependency as long as the function is not explicitly used in some production code.

For example if `ChecksUser` is explicitly called elsewhere in the production code the binary is 1MB bigger and all the test tooling seems to be embedded in the binary:
```
$ strings elastic-operator|grep -i 'testing\.'
*testing.T
**testing.T
*[]*testing.T
*testing.common
**testing.common
*testing.matcher
*func(*testing.T)
*testing.indenter
...
```

If the function `ChecksUser` is shared by some tests but not in the production code _(like in this PR)_: 
```
$ strings elastic-operator|grep -i 'testing\.'
testing.init
/usr/local/Cellar/go/1.11.4/libexec/src/testing/testing.go
$
```
_it is the same result with the code on the current branch btw_

Reply:
👍 Nice analysis!